agent/cache-types/connect_ca_leaf.go

```diff
@@ -679,6 +693,7 @@ type ConnectCALeafRequest struct {
 	Service        string              // Service name, not ID
 	Agent          string              // Agent name, not ID
 	Kind           structs.ServiceKind // only mesh-gateway for now
+	Server         bool
```

Comment:
Can we leave a short doc on what fields are mutually exclusive? It's not immediately obvious here without looking at the switch-case

Reply:
Yep, done